core/src/apps/cardano/sign_tx.py

```diff
@@ -468,9 +473,14 @@ async def _show_standard_tx(
 
     if not is_network_id_verifiable:
         await show_warning_tx_network_unverifiable(ctx)
+
     total_amount = await _show_outputs(ctx, keychain, msg)
 
     for certificate in msg.certificates:
+        if not SCHEMA_STAKING.match(certificate.path):
```

Comment:
there's a pattern worth abstracting:
```
if not SOME_SCHEMA.match(path):
    if is_strict:
        raise wire.DataError("Invalid " + path_name)
    else:
        await _show_warning(ctx, path, path_name)
```

Reply:
Updated. https://github.com/trezor/trezor-firmware/pull/1423/commits/b618717511081a591a86dab20c42a5a083a7b4d6 At first I wanted to place the function into `utils.py` but there was a problem with circular dependencies and rather than locally importing `layout.py` I moved it to `sign_tx.py` since it's used only there anyways.